datarepo-clienttests/src/main/java/utils/KubernetesClientUtils.java

```diff
@@ -0,0 +1,264 @@
+        new InputStreamReader(new FileInputStream(kubeConfigPath), StandardCharsets.UTF_8);
+    KubeConfig kubeConfig = KubeConfig.loadKubeConfig(filereader);
+
+    // get a refreshed SA access token and its expiration time
+    GoogleCredentials applicationDefaultCredentials =
+        AuthenticationUtils.getApplicationDefaultCredential();
+    AccessToken accessToken = AuthenticationUtils.getAccessToken(applicationDefaultCredentials);
+    Instant tokenExpiration = accessToken.getExpirationTime().toInstant();
+    String expiryUTC = tokenExpiration.atZone(ZoneOffset.UTC).format(DateTimeFormatter.ISO_INSTANT);
+
+    // USERS: build list of one user, the SA
+    LinkedHashMap<String, Object> authConfigSA = new LinkedHashMap<>();
+    authConfigSA.put("access-token", accessToken.getTokenValue());
+    authConfigSA.put("expiry", expiryUTC);
+
+    LinkedHashMap<String, Object> authProviderSA = new LinkedHashMap<>();
+    authProviderSA.put("name", "gcp");
+    authProviderSA.put("config", authConfigSA);
+
+    LinkedHashMap<String, Object> userSA = new LinkedHashMap<>();
+    userSA.put("auth-provider", authProviderSA);
+
+    LinkedHashMap<String, Object> userWrapperSA = new LinkedHashMap<>();
+    userWrapperSA.put("name", server.clusterName);
+    userWrapperSA.put("user", userSA);
+
+    ArrayList<Object> usersList = new ArrayList<>();
+    usersList.add(userWrapperSA);
+
+    // CONTEXTS: build list of one context, the specified cluster
+    LinkedHashMap<String, Object> context = new LinkedHashMap<>();
+    context.put("cluster", server.clusterName);
+    context.put(
+        "user", server.clusterName); // when is the user ever different from the cluster name?
+
+    LinkedHashMap<String, Object> contextWrapper = new LinkedHashMap<>();
+    contextWrapper.put("name", server.clusterName);
+    contextWrapper.put("context", context);
+
+    ArrayList<Object> contextsList = new ArrayList<>();
+    contextsList.add(contextWrapper);
+
+    // CLUSTERS: use the cluster list read in from the kubeconfig file, because I can't figure out
+    // how to get the certificate-authority-data and server address for the cluster via the Java
+    // client library, only with gcloud
+    ArrayList<Object> clusters = kubeConfig.getClusters();
+
+    // build the config object, replacing the contexts and users lists from the kubeconfig file with
+    // the ones constructed programmatically above
+    kubeConfig = new KubeConfig(contextsList, clusters, usersList);
+    kubeConfig.setContext(server.clusterName);
+
+    // build the client object from the config
+    ApiClient client = ClientBuilder.kubeconfig(kubeConfig).build();
+
+    // set the global default client to the one created above because the CoreV1Api and AppsV1Api
+    // constructors get the client object from the global configuration
+    Configuration.setDefaultApiClient(client);
+
+    kubernetesClientCoreObject = new CoreV1Api();
+    kubernetesClientAppsObject = new AppsV1Api();
```

Comment:
It looks like there is an option to take the "client" variable as input to the CoreV1Api declaration. 
`kubernetesClientCoreObject = new CoreV1Api(client);`
But, clearly, it works without doing that… Do you know why?

Reply:
Yes, if you don't pass the client, it just uses Configuration.getDefaultApiClient(), which I've set in the line above.